lib/stm32/f4/pwr.c

```diff
@@ -36,11 +36,9 @@
 
 #include <libopencm3/stm32/pwr.h>
 
-void pwr_set_vos_scale(enum pwr_vos_scale scale)
+void pwr_set_vos_scale(int8_t scale)
 {
-	if (scale == PWR_SCALE1) {
-		PWR_CR |= PWR_CR_VOS;
-	} else if (scale == PWR_SCALE2) {
-		PWR_CR &= PWR_CR_VOS;
-	}
+	uint32_t reg = PWR_CR & ~(PWR_CR_VOS_MASK << PWR_CR_VOS_SHIFT);
+	reg |= (scale & PWR_CR_VOS_MASK) << PWR_CR_VOS_SHIFT;
+	PWR_CR = reg;
```

Comment:
it does make less code doesn't it...  perhaps the enums with defined values?

Reply:
either works although when they change it to 3 bits the enum needs to be reworked. The defined values allows code transparency while the changes happen at the loss of compiler ability to detect mis-matched arguments. This function isn't called in a lot of code (certainly on the F4 side)